### PR TITLE
Model import

### PR DESCRIPTION
**Pull request by YahimaB**

### CODE REVIEW

File: Engine/Assets/Mesh.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+using LinearAlgebra;
+using SharpDX;
+using SharpDX.Direct3D;
+using SharpDX.Direct3D11;
+using SharpDX.DXGI;
+using Buffer = SharpDX.Direct3D11.Buffer;
+
+namespace Engine
+{
+    public class Mesh : BaseAsset //TODO: rename to model?
```
Comment (Notrum666) [inline Engine/Assets/Mesh.cs:12]: yes

File: Engine/Assets/Mesh.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                foreach (Primitive primitive in Primitives)
+                    primitive.Dispose(disposing);
+            }
+            disposed = true;
+
+            base.Dispose(disposing);
+        }
+    }
+
+    public class Primitive : IDisposable //TODO: rename to mesh?
```
Comment (Notrum666) [inline Engine/Assets/Mesh.cs:41]: yes

File: Engine/Assets/Material.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                    throw new ArgumentNullException("Emissive", "Texture can't be null.");
+                emissive = value;
+            }
+        }
+
+        private bool disposed = false;
+
+        #region Legacy
+
+        public static Material Default()
```
Comment (Notrum666) [inline Engine/Assets/Material.cs:78]: change into readonly property so object is not re-created every time

**Review by Notrum666 — CHANGES_REQUESTED** [review]

File: SharpDXtest/AssetsImport/AssetMeta.cs [hunk 4]
```diff
@@ -8,6 +10,9 @@ public class AssetMeta
 
         public string Guid = System.Guid.NewGuid().ToString("N");
         public DateTime ImportDate = DateTime.MinValue;
+        
+        [YamlMember(DefaultValuesHandling = DefaultValuesHandling.OmitEmptyCollections)]
+        public Dictionary<(Type, string), string> SubAssets = new Dictionary<(Type, string), string>();
 
         public object ImportSettings = null;
```
[the remark this answers is not in the document]
Comment (YahimaB) [reply SharpDXtest/AssetsImport/AssetMeta.cs:17]: Temp solution - more options

**Review by Notrum666 — APPROVED** [review]

**Pull request merged**